modules/core/src/main/java/org/apache/ignite/internal/binary/RawBytesObjectReader.java

```diff
@@ -0,0 +1,335 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.binary;
+
+import org.apache.ignite.binary.BinaryObjectException;
+import org.apache.ignite.internal.binary.streams.BinaryInputStream;
+import org.apache.ignite.internal.binary.streams.BinaryOutputStream;
+
+/** */
+public class RawBytesObjectReader implements BinaryPositionReadable {
+    /** */
+    private final BinaryInputStream in;
+
+    /** */
+    public RawBytesObjectReader(BinaryInputStream in) {
+        this.in = in;
+    }
+
+    /** */
+    public byte[] readObject() {
+        int startPos = in.position();
+
+        skipObject();
+
+        int endPos = in.position();
+
+        in.position(startPos);
+
+        return in.readByteArray(endPos - startPos);
+    }
+
+    /** */
+    public void readObject(BinaryOutputStream out) {
```

Comment:
It's not a good name for this method (and following methods with `BinaryOutputStream out` parameter), perhaps something like `copyObject` is better.